omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ServerLogResource1_8.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.MapProperty;
+import org.openmrs.module.webservices.helper.ServerLogActionWrapper;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Listable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link Resource} for ServerLogController, supporting standard CRUD operations
+ */
+@Resource(name = RestConstants.VERSION_1 + "/serverlog", supportedClass = ServerLogActionWrapper.class, supportedOpenmrsVersions = {
+        "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*" })
+public class ServerLogResource1_8 extends BaseDelegatingResource<ServerLogActionWrapper> implements Listable {
+	
+	private ServerLogActionWrapper serverLogActionWrapper = new ServerLogActionWrapper();
+	
+	public void setServerLogActionWrapper(ServerLogActionWrapper serverLogActionWrapper) {
+		this.serverLogActionWrapper = serverLogActionWrapper;
+	}
+	
+	@Override
+	public SimpleObject getAll(RequestContext context) throws ResponseException {
+		SimpleObject rest = new SimpleObject();
+		rest.put("serverLog", serverLogActionWrapper.getServerLogs());
```

Comment:
Do you need to put the logs in single field? I'd Imagine that you just need to a return a list of server logs

Reply:
I thought to get that list of logs from "serverlog" field. Anyway, returning that list directly to the resource also a good point. Thanks @wluyima

Reply:
Since the server logs will return nearly 100 lines, I kept those inside a single key to reduce the complexity for now. We can add some other keys based later as well.